dashboard/app/models/levels/level.rb

```diff
@@ -33,6 +33,16 @@ class Level < ActiveRecord::Base
   has_many :level_sources
   has_many :hint_view_requests
 
+  # We store parent-child relationships in a self-referential join table.
+  # In order to define a has_many / through relationship in both directions,
+  # we must define two separate associations to the same join table.
+
+  has_many :levels_parent_levels, class_name: 'ParentLevelsChildLevel', foreign_key: :child_level_id
```

Comment:
Is `:levels_parent_levels` needed separately from `:parent_levels`? If not, is it not possible syntactically to combine them?

Reply:
we could effectively combine them by using `has_and_belongs_to_many` instead. If we did, there would be no way to access the `kind` or `position` parameters, though. See https://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many